src/SimpleNotePad.java

```diff
@@ -1,120 +1,43 @@
+import actions.*;
+import utilities.*;
+
 import java.awt.Dimension;
-import java.awt.Graphics;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
-import java.awt.print.PageFormat;
-import java.awt.print.Printable;
-import java.awt.print.PrinterException;
-import java.awt.print.PrinterJob;
-import java.io.File;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.io.PrintWriter;
-import javax.swing.JFileChooser;
 import javax.swing.JFrame;
 import javax.swing.JMenu;
 import javax.swing.JMenuBar;
-import javax.swing.JMenuItem;
-import javax.swing.JOptionPane;
 import javax.swing.JScrollPane;
 import javax.swing.JTextPane;
-import javax.swing.text.Position;
-import javax.swing.text.StyledDocument;
-public class SimpleNotePad extends JFrame implements ActionListener{
```

Comment:
I think the abstraction of implementing ActionListener is the key to refactoring this document. I was hesitant remove this from SimpleNotePad because I was unsure if it is within best practices guidelines. After looking over this code I think I this is absolutely the best way to handle this.